File: .ci/doc/templates/catch.tpl.java
try {
  AbstractProtocol protocol;
  if (System.getenv("SNIPPET_PROTOCOL") != null && System.getenv("SNIPPET_PROTOCOL").equals("http")) {
    protocol = new Http("kuzzle");
  } else {
    protocol = new WebSocket("kuzzle");
  }
  kuzzle = new Kuzzle(protocol);
  kuzzle.connect();
  [snippet-code]
  System.out.println("Error");
} catch (Exception e) {
  e.printStackTrace();
} finally {
  if (kuzzle != null) {
    kuzzle.disconnect();
  }
}

File: .ci/doc/templates/catch.tpl.kt
import io.kuzzle.sdk.Kuzzle
import io.kuzzle.sdk.protocol.*

import java.util.concurrent.ExecutionException
import io.kuzzle.sdk.coreClasses.responses.Response

fun main() {
  val protocol: AbstractProtocol;
  if (System.getenv("SNIPPET_PROTOCOL") == "http") {
    protocol = Http("kuzzle")
  } else {
    protocol = WebSocket("kuzzle")
  }
  val kuzzle = Kuzzle(protocol).apply {
    connect()
  }
  try {
    [snippet-code]
    println("Error")
  } catch (e: Exception) {
    e.printStackTrace()
  } finally {
    kuzzle.disconnect()
  }
}
